### ISSUE TITLE
Rules for const and async acts as mutually exclusive

### ISSUE THREAD

**Issue by nwinger** [eslint/eslint#9596]
**Tell us about your environment**

* **ESLint Version: 2.9.0**
* **Node Version: 8.0.0**
* **npm Version: 5.5.1**

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```{
  "parserOptions": {
    "ecmaVersion": 2017
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const mongoose = require("mongoose");
const Store = mongoose.model("Store");

exports.homePage = (req, res) => {
  console.log(req.name);
  res.render("index");
};

exports.addStore = (req, res) => {
  res.render("editStore", { title: "Add Store" });
};

exports.createStore = async (req, res) => {
  const store = new Store(req.body);
  await store.save();
};
```

```bash
Running eslint in VSCode via dbaeumer.vscode-eslint
```

**What did you expect to happen?**
No linting errors to appear
**What actually happened? Please include the actual, raw output from ESLint.**
When running ecmaVersion 2017 in the .eslintrc file in VSCode, const and async linting errors acts as mutually exclusive. ecmaVersion 6 allows for using const, but not async. Async reports "Parsing error: Unexpected token =>". ecmaVersion 2017 allows for using async, but not const. const reports "Parsing error: ecmaVersion must be 3, 5, 6, or 7."

**Comment by platinumazure** [comment]
Are you really on ESLint 1.4.2? Or is that the VSCode version?

ESLint 1.4.2 is quite old, and it is possible that it does not ship with a version of espree (default parser) that supports ecmaVersion 8 or 2017 (which mean the same thing). If you're using ESLint 1.4.2, my recommendation would be to try upgrading.

**Comment by nwinger** [comment]
@platinumazure Sorry - my mistake! ``eslint -v`` yields v2.9.0.
I'm running it via this plugin in VSCode: https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint

**Comment by Aladdin-ADD** [comment]
I think eslint v2.9.0 does not support ES2017, you can upgrade to latest eslint to fix it.

**Comment by nwinger** [comment]
Upgraded to eslint 4.11.0 globally, and the problem got resolved. Closing the issue :)
